UI/homepage.py

```diff
@@ -4,6 +4,67 @@
 # Initialize PyGame
 pygame.init()
 
+
+class Checkbox:
+    def __init__(self, surface, x, y, idnum, color=(230, 230, 230),
+        caption="", outline_color=(0, 0, 0), check_color=(0, 0, 0),
+        font_size=22, font_color=(0, 0, 0), 
+        text_offset=(28, 1), font='Arial'):
+        self.surface = surface
+        self.x = x
+        self.y = y
+        self.color = color
+        self.caption = caption
+        self.oc = outline_color
+        self.cc = check_color
+        self.fs = font_size
+        self.fc = font_color
+        self.to = text_offset
+        self.ft = font
+
+        #identification for removal and reorganization
+        self.idnum = idnum
+
+        # checkbox object
+        self.checkbox_obj = pygame.Rect(self.x, self.y, 12, 12)
+        self.checkbox_outline = self.checkbox_obj.copy()
+
+        # variables to test the different states of the checkbox
+        self.checked = False
+
+    def _draw_button_text(self):
+        self.font = pygame.font.SysFont(self.ft, self.fs)
+        self.font_surf = self.font.render(self.caption, True, self.fc)
```

Comment:
your previous self. variables are all abbreviated, but these aren't, so as a note for future coding it's good to stay consistent, and also if you abbreviate names, it's good to make sure that they're not so abbreviated that they can confuse your readers